### ISSUE TITLE
Using "Spatial" extension to read excel is giving error with versions 0.9.0 and above. It throws "NotImplementedException: Not implemented Error: GDAL Error (6): Adding too many columns to too many existing features". The same code worked well with duck db0.8.1

### ISSUE THREAD

**Issue by debottamroychowdhury** [duckdb/duckdb-spatial#230]
### What happens?

Reading this line gives me below error with latest versions of duck db

NotImplementedException: Not implemented Error: GDAL Error (6): Adding too many columns to too many existing features

### To Reproduce

CREATE OR REPLACE TABLE 'DummyTable' AS SELECT * FROM ST_Read('file_name', layer='Sheet1', open_options=['HEADER=FORCE'])

### OS:

Windows

### DuckDB Version:

0.9.0

### DuckDB Client:

Python 3.10

### Full Name:

Debottam

### Affiliation:

Personal

### Have you tried this on the latest `main` branch?

I have tested with a main build

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

**Comment by Maxxen** [comment]
Hi! Thanks for filing this issue. Does the same problem occur if you the latest nightly version of spatial? You can get it for DuckDB v0.9.2 by running:

```sql
FORCE INSTALL spatial FROM 'http://nightly-extensions.duckdb.org';
```

**Comment by debottamroychowdhury** [comment]
> the same problem occur if you the latest nightly version of spatial? You can get it for DuckDB v0.9.2 by running:

Yes, tried to do FORCE INSTALL spatial FROM 'http://nightly-extensions.duckdb.org'; in the code, but same issue still persists.

**Comment by Maxxen** [comment]
Could you share the excel file?

**Comment by debottamroychowdhury** [comment]
> Could you share the excel file? You can email me at [max@duckdblabs.com](mailto:max@duckdblabs.com) if you don't want to share it publicly.

I think you have just identified the actual issue. I just discovered the same excel file which worked well with same code in duck db 0.8.1 Is not working with duck db 0.9.x. I copied the contents of the excel file to a new excel and then it worked well in 0.9.x also. Not exactly sure about the reason for this issue as the excel file seems good to me.
But I noticed, whatever header options I provide in below, be it AUTO/DISABLE/FORCE, it is kind of not considering this and always adding one default header as field 0, field 1..field n
open_options=['HEADER=FORCE']

**Comment by debottamroychowdhury** [comment]
> Could you share the excel file? You can email me at [max@duckdblabs.com](mailto:max@duckdblabs.com) if you don't want to share it publicly.

I think you have just identified the actual issue. I just discovered the same excel file which worked well with same code in duck db 0.8.1 Is not working with duck db 0.9.x. I copied the contents of the excel file to a new excel and then it worked well in 0.9.x also. 

The reason for this issue is clear now, in my excel, we had data in all rows of the columns (A to D) + there were some data in few rows towards the end of the sheet (for column G to column H). Such an excel is not working and is throwing this error with Duck DB 0.9.x versions. Please note, this same excel worked well with 0.8.1

**Comment by debottamroychowdhury** [comment]
> Could you share the excel file? You can email me at [max@duckdblabs.com](mailto:max@duckdblabs.com) if you don't want to share it publicly.

Kindly also note, if we put the data of column G to H towards the start of the excel (randomly say from row 6 onwards), then it works well. The issue only occurs when such data is towards the end of the sheet.

**Comment by Maxxen** [comment]
Hello!

As of DuckDB 1.2.0 the duckdb [excel extension](https://duckdb.org/docs/extensions/excel.html#reading-xlsx-files) provides support for reading and writing xlsx files with greater efficiency and many more options. Therefore we're going to deprecate this feature in the spatial extension in the future. I would recommend that you try out the excel extension, and if you run into any problems please open an issue over at the [duckdb-excel repository](https://github.com/duckdb/duckdb-excel).
